### PR TITLE
htslib: migrate from homebrew-science

### PR DESCRIPTION
**Pull request by iMichka**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by iMichka** [comment]
Companion PR: https://github.com/Homebrew/homebrew-science/pull/6366

File: Formula/htslib.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+class Htslib < Formula
+  desc "C library for high-throughput sequencing data formats"
+  homepage "http://www.htslib.org/"
+  url "https://github.com/samtools/htslib/releases/download/1.6/htslib-1.6.tar.bz2"
+  sha256 "9588be8be0c2390a87b7952d644e7a88bead2991b3468371347965f2e0504ccb"
+  revision 1
+
+  head "https://github.com/samtools/htslib.git"
+
+  depends_on "xz"
```
Comment (ilovezfs) [inline Formula/htslib.rb]: are we sure this is needed?

File: Formula/htslib.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    pkgshare.install "test"
+  end
+
+  test do
+    sam = pkgshare/"test/ce#1.sam"
+    assert_match "SAM", shell_output("#{bin}/htsfile #{sam}")
+    system "#{bin}/bgzip -c #{sam} > sam.gz"
+    assert File.exist?("sam.gz")
+    system "#{bin}/tabix", "-p", "sam", "sam.gz"
+    assert File.exist?("sam.gz.tbi")
```
Comment (ilovezfs) [inline Formula/htslib.rb]: assert_predicate and a message

File: Formula/htslib.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    system "./configure", "--enable-libcurl"
+    system "make", "install", "prefix=#{prefix}"
+    pkgshare.install "test"
+  end
+
+  test do
+    sam = pkgshare/"test/ce#1.sam"
+    assert_match "SAM", shell_output("#{bin}/htsfile #{sam}")
+    system "#{bin}/bgzip -c #{sam} > sam.gz"
+    assert File.exist?("sam.gz")
```
Comment (ilovezfs) [inline Formula/htslib.rb]: ditto

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by ilovezfs** [comment]
```
Error: 2 problems in 1 formula
htslib:
  * Use `assert_predicate <path_to_file>, :exist?` instead of `assert File.exist?`
  * Use `assert_predicate <path_to_file>, :exist?` instead of `assert File.exist?`
```

File: Formula/htslib.rb (hunk 1 above)
Comment (iMichka) [reply Formula/htslib.rb]: Just tested, it is not. I'll remove it.

**Comment by iMichka** [comment]
```
configure: error: liblzma development files not found

The CRAM format may use LZMA2 compression, which is implemented in HTSlib
by using compression routines from liblzma <http://tukaani.org/xz/>.

Building HTSlib requires liblzma development files to be installed on the
build machine; you may need to ensure a package such as liblzma-dev (on Debian
or Ubuntu Linux), xz-devel (on RPM-based Linux distributions or Cygwin), or
xz (via Homebrew on macOS) is installed; or build XZ Utils from source.
```

Actually `depends_on "xz"` is needed.

File: Formula/htslib.rb (hunk 1 above)
Comment (ilovezfs) [reply Formula/htslib.rb]: probably just pass --disable-lzma

**Comment by JCount** [comment]
@iMichka might it be worth adding `depends_on "xz"` to the formula rather than disabling `LZMA2`? It would be consistent with the original `homebrew-science` formula after all.

**Comment by iMichka** [comment]
@ilovezfs asked me to remove it. I do not care if it's there or not. I guess there is no harm to leave it.

**Comment by JCount** [comment]
Hmm, I'll talk to him.

File: Formula/htslib.rb (hunk 1 above)
Comment (JCount) [reply Formula/htslib.rb]: It isn't strictly required, but it is a fairly small dependency that may increase the utility of this formula.

**Comment by JCount** [comment]
@BrewTestBot test this please

**Comment by iMichka** [comment]
+1 on the revision bump, forgot that. Good to go for me.

File: Formula/htslib.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  url "https://github.com/samtools/htslib/releases/download/1.6/htslib-1.6.tar.bz2"
+  sha256 "9588be8be0c2390a87b7952d644e7a88bead2991b3468371347965f2e0504ccb"
+  revision 1
+
+  head "https://github.com/samtools/htslib.git"
+
+  depends_on "xz"
+
+  def install
+    system "./configure", "--enable-libcurl"
```
Comment (ilovezfs) [inline Formula/htslib.rb]: can we not set `--prefix=#{prefix}` here?

**Review by ilovezfs — CHANGES_REQUESTED** [review]

**Comment by iMichka** [comment]
I moved the `--prefix` to the `configure` step and squashed all the commits.

**Comment by iMichka** [comment]
Good now? :)

File: Formula/htslib.rb [hunk 5]
```diff
@@ -0,0 +1,27 @@
+class Htslib < Formula
+  desc "C library for high-throughput sequencing data formats"
+  homepage "http://www.htslib.org/"
+  url "https://github.com/samtools/htslib/releases/download/1.6/htslib-1.6.tar.bz2"
+  sha256 "9588be8be0c2390a87b7952d644e7a88bead2991b3468371347965f2e0504ccb"
+  revision 1
+
+  head "https://github.com/samtools/htslib.git"
```
Comment (ilovezfs) [inline Formula/htslib.rb]: head needs autoreconf run, and the Autotools dependencies, but let's just drop it

File: Formula/htslib.rb [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+  def install
+    system "./configure",
```
Comment (ilovezfs) [inline Formula/htslib.rb]: this should be formatted
```
    system "./configure", "--prefix=#{prefix}",
                          "--enable-libcurl",
                          "--disable-lzma"
```

**Review by ilovezfs — CHANGES_REQUESTED** [review]
I'm fine with leaving lzma enabled if you want

File: Formula/htslib.rb (hunk 6 above)
Comment (ilovezfs) [reply Formula/htslib.rb]: Also, it looks like --enable-libcurl is already the default

**Review by ilovezfs — APPROVED** [review]

**Pull request closed without merging**